src/core/getProfits.ts

```diff
@@ -85,6 +125,19 @@ export async function getProfits({
         protocolTokenAddresses: Object.keys(endPositionValues), // endPositionValues is indexed by tokenId ?? protocolTokenAddress
         tokenIds: Object.keys(endPositionValues), // endPositionValues is indexed by tokenId ?? protocolTokenAddress
       })
+      .then(async (result) => {
+        await resolveUnderlyings(
+          adapter,
+          fromBlock,
+          result,
+          (underlyingToken, protocolToken, underlyingRateRaw) => {
```

Comment:
I think in this file the function is always the same yeah? Do you mind extracting the function?

e.g. rateCalculation